homeassistant/components/switch/rachio.py

```diff
@@ -4,227 +4,226 @@
 
-REQUIREMENTS = ['rachiopy==0.1.2']
+DEPENDENCIES = ['rachio']
 
 _LOGGER = logging.getLogger(__name__)
 
+# Manual run length
 CONF_MANUAL_RUN_MINS = 'manual_run_mins'
-
-DATA_RACHIO = 'rachio'
-
 DEFAULT_MANUAL_RUN_MINS = 10
 
-MIN_UPDATE_INTERVAL = timedelta(seconds=30)
-MIN_FORCED_UPDATE_INTERVAL = timedelta(seconds=1)
-
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_MANUAL_RUN_MINS, default=DEFAULT_MANUAL_RUN_MINS):
         cv.positive_int,
 })
 
+ATTR_ZONE_SUMMARY = 'Summary'
+ATTR_ZONE_NUMBER = 'Zone number'
+
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Rachio switches."""
-    from rachiopy import Rachio
+    manual_run_time = timedelta(minutes=config.get(CONF_MANUAL_RUN_MINS))
+    _LOGGER.info("Rachio run time is %s", str(manual_run_time))
 
-    # Get options
-    manual_run_mins = config.get(CONF_MANUAL_RUN_MINS)
-    _LOGGER.debug("Rachio run time is %d min", manual_run_mins)
+    # Add all zones from all controllers as switches
+    devices = []
+    for controller in hass.data[DOMAIN].controllers:
+        devices.append(RachioStandbySwitch(hass, controller))
 
-    access_token = config.get(CONF_ACCESS_TOKEN)
+        for zone in controller.list_zones():
+            devices.append(RachioZone(hass, controller, zone, manual_run_time))
 
-    # Configure API
-    _LOGGER.debug("Configuring Rachio API")
-    rachio = Rachio(access_token)
+    add_devices(devices)
+    _LOGGER.info("%d Rachio switch(es) added", len(devices))
 
-    person = None
-    try:
-        person = _get_person(rachio)
-    except KeyError:
-        _LOGGER.error(
-            "Could not reach the Rachio API. Is your access token valid?")
-        return
 
-    # Get and persist devices
-    devices = _list_devices(rachio, manual_run_mins)
-    if not devices:
-        _LOGGER.error(
-            "No Rachio devices found in account %s", person['username'])
-        return
+class RachioSwitch(SwitchDevice):
+    """Represent a Rachio state that can be toggled."""
 
-    hass.data[DATA_RACHIO] = devices[0]
+    def __init__(self, controller, poll=True):
+        """Initialize a new Rachio switch."""
+        self._controller = controller
 
-    if len(devices) > 1:
-        _LOGGER.warning("Multiple Rachio devices found in account, "
-                        "using %s", hass.data[DATA_RACHIO].device_id)
-    else:
-        _LOGGER.debug("Found Rachio device")
+        if poll:
+            self._state = self._poll_update()
+        else:
+            self._state = None
 
-    hass.data[DATA_RACHIO].update()
-    add_devices(hass.data[DATA_RACHIO].list_zones())
+    @property
+    def __str__(self):
+        """Return a string representation of this switch."""
+        return self.name
 
+    @property
+    def name(self) -> str:
+        """Get a name for this switch."""
+        return "Switch on {}".format(self._controller.name)
```

Comment:
Doesn't every controller have 3 switches?

Reply:
For now, yes. But I want it to be easy to add more switchable options in the future.